internal/repository/skr/webhook/skrwebhook_repo_test.go

```diff
@@ -0,0 +1,405 @@
+		getCallCount := 0
+		mockClient := &mockSkrClient{
+			getFunc: func(ctx context.Context,
+				key client.ObjectKey,
+				obj client.Object,
+				opts ...client.GetOption,
+			) error {
+				getCallCount++
+				// First resource exists
+				if key.Name == "skr-webhook" {
+					return nil
+				}
+				return apierrors.NewNotFound(schema.GroupResource{}, key.Name)
+			},
+		}
+
+		clientCache := &mockSkrClientCache{client: mockClient}
+		repo := webhook.NewResourceRepository(clientCache, remoteSyncNamespace, baseResources)
+
+		exists, err := repo.ResourcesExist(kymaName)
+
+		require.NoError(t, err)
+		assert.True(t, exists)
+		// Should short-circuit after finding first resource
+		assert.LessOrEqual(t, getCallCount, len(baseResources)+2) // +2 for generated resources
+	})
+
+	t.Run("returns false when no resources exist", func(t *testing.T) {
+		mockClient := &mockSkrClient{
+			getFunc: func(ctx context.Context,
+				key client.ObjectKey,
+				obj client.Object,
+				opts ...client.GetOption,
+			) error {
+				return apierrors.NewNotFound(schema.GroupResource{}, key.Name)
+			},
+		}
+
+		clientCache := &mockSkrClientCache{client: mockClient}
+		repo := webhook.NewResourceRepository(clientCache, remoteSyncNamespace, baseResources)
+
+		exists, err := repo.ResourcesExist(kymaName)
+
+		require.NoError(t, err)
+		assert.False(t, exists)
+	})
+
+	t.Run("returns true when generated resource exists", func(t *testing.T) {
+		mockClient := &mockSkrClient{
+			getFunc: func(ctx context.Context,
+				key client.ObjectKey,
+				obj client.Object,
+				opts ...client.GetOption,
+			) error {
+				// Only ValidatingWebhookConfiguration exists
+				if key.Name == skrwebhookresources.SkrResourceName {
+					unstr, ok := obj.(*unstructured.Unstructured)
+					if ok && unstr.GetKind() == "ValidatingWebhookConfiguration" {
+						return nil
+					}
+				}
+				return apierrors.NewNotFound(schema.GroupResource{}, key.Name)
+			},
+		}
+
+		clientCache := &mockSkrClientCache{client: mockClient}
+		repo := webhook.NewResourceRepository(clientCache, remoteSyncNamespace, baseResources)
+
+		exists, err := repo.ResourcesExist(kymaName)
+
+		require.NoError(t, err)
+		assert.True(t, exists)
+	})
+
+	t.Run("returns error when Get fails with non-NotFound error", func(t *testing.T) {
+		expectedErr := errors.New("API server error")
+		mockClient := &mockSkrClient{
+			getFunc: func(ctx context.Context,
+				key client.ObjectKey,
+				obj client.Object,
+				opts ...client.GetOption,
+			) error {
+				return expectedErr
+			},
+		}
+
+		clientCache := &mockSkrClientCache{client: mockClient}
+		repo := webhook.NewResourceRepository(clientCache, remoteSyncNamespace, baseResources)
+
+		exists, err := repo.ResourcesExist(kymaName)
+
+		require.Error(t, err)
+		assert.False(t, exists)
+		assert.Contains(t, err.Error(), "failed to check resource")
+	})
+
+	t.Run("short-circuits on first found resource", func(t *testing.T) {
```

Comment:
What is this test adding on to pof the `returns true when at least one resource exists`?